web/src/components/PlanWorkoutModalForm/PlanWorkoutModalForm.js

```diff
@@ -0,0 +1,210 @@
+          <PdlSelect
+            form={form}
+            id="activityId"
+            label="Activity"
+            defaultValue={planWorkout?.activityId}
+            validations={{ required: true }}
+          >
+            <ActivityOptions activities={props.activities} />
+          </PdlSelect>
+
+          <Spacer mx={5} />
+
+          <PdlCheckbox
+            form={form}
+            id="isKeyWorkout"
+            label="Key workout?"
+            defaultValue={planWorkout?.isKeyWorkout}
+          />
+        </Flex>
+
+        <Divider my={5} />
+
+        <Heading as="h3" size="md" my={3}>
+          Goals
+        </Heading>
+
+        <Flex flexDirection="row">
+          <PdlInput
+            form={form}
+            id="targetMiles"
+            label="Target miles"
+            defaultValue={planWorkout?.targetMiles}
+          />
+
+          <Spacer mx={5} />
+
+          <PdlInput
+            form={form}
+            id="targetTimeInMinutes"
+            label="Target time (minutes)"
+            defaultValue={planWorkout?.targetTimeInMinutes}
+          />
+        </Flex>
+
+        <Spacer my={5} />
+
+        <PdlInput
+          form={form}
+          id="targetNotes"
+          label="Target notes"
+          defaultValue={planWorkout?.targetNotes}
+        />
+
+        <Divider my={5} />
+
+        <Button
+          colorScheme="green"
+          isLoading={isSubmitting || props.loading}
+          type="submit"
+          disabled={isSubmitting || props.loading}
+        >
+          Save
+        </Button>
+      </form>
+    </div>
+  )
+}
+
+const PdlInput = ({ form, id, label, defaultValue, validations }) => {
+  const {
+    register,
+    formState: { errors },
+  } = form
+
+  return (
+    <PdlField errors={errors} id={id} label={label}>
+      <Input
+        {...register(id, validations)}
+        placeholder={label}
+        defaultValue={defaultValue}
+      />
+    </PdlField>
+  )
+}
+
+const PdlSelect = ({
+  children,
+  form,
+  id,
+  label,
+  defaultValue,
+  validations,
+}) => {
+  const {
+    register,
+    formState: { errors },
+  } = form
+
+  return (
+    <PdlField errors={errors} id={id} label={label}>
+      <Select {...register(id, validations)} defaultValue={defaultValue}>
+        {children}
+      </Select>
+    </PdlField>
+  )
+}
+
+const PdlCheckbox = ({ form, id, label, defaultValue }) => {
+  const {
+    register,
+    formState: { errors },
+  } = form
+
+  return (
+    <PdlField errors={errors} id={id} label={label}>
+      <Checkbox {...register(id)} defaultChecked={defaultValue}>
+        {label}
+      </Checkbox>
+    </PdlField>
+  )
+}
+
+const PdlField = ({ children, errors, id, label }) => {
+  return (
+    <FormControl isInvalid={errors[id]}>
+      <FormLabel htmlFor={id}>{label}</FormLabel>
+      {children}
+
+      <FormErrorMessage>
+        {/* TODO: a more robust solution for error messages */}
+        {errors[id] && `${label} is required`}
```

Comment:
This only works right now because the only type of validation I have is `required`...so if there's _any_ error, it must be of that type.